components/sp-solutions/org.wso2.carbon.sp.solutions.apim.analytics/components/org.wso2.sp.extension.siddhi.io.mgwfile/src/main/java/org/wso2/extension/siddhi/io/mgwfile/MGWFileDataRetrieverThreadFactory.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.extension.siddhi.io.mgwfile;
+
+import java.util.concurrent.ThreadFactory;
+
+/**
+ * This is the Thread Factory used to create the threads which reads the files from the DB and map them to streams
+ */
+public class MGWFileDataRetrieverThreadFactory implements ThreadFactory {
```

Comment:
During the review, the ThreadFactory implementation we had was different and this is a more simplified version of it. According to my notes, we did not decide to remove the factory but rather questioned about the implementation of it. 

By default, the thread factory will create 3 threads to read from the database. Which means 3 files at a time. This is also configurable. When the unread files in the DB are large, I think this can be useful. Do you see an issue?

Reply:
I was wondering whether we really need to extend `ThreadFactory` rather than using executor service. During the code review also, we discussed that: whether we really needed it. I was wondering what is the outcome of that discussion: whether to keep it or not. Anyway I see no issue keeping this.